nimare/meta/utils.py

```diff
@@ -337,30 +388,43 @@ def compute_kda_ma(
     cube = np.vstack([row.ravel() for row in np.mgrid[xx, yy, zz]])
     kernel = cube[:, np.sum(np.dot(np.diag(vox_dims), cube) ** 2, 0) ** 0.5 <= r]
 
-    # Preallocate coords array, np.concatenate is too slow
-    n_coords = ijks.shape[0] * kernel.shape[1]  # = n_peaks * n_voxels
-    coords = np.zeros((4, n_coords), dtype=int)
+    def _convolve_sphere(kernel, peaks):
+        # Convolve spheres
```

Comment:
```suggestion
        """Convolve peaks with a spherical kernel.

        Parameters
        -----------
        kernel : 2D numpy.ndarray
            IJK coordinates of a sphere, relative to a central point (not the brain template).
        peaks : 2D numpy.ndarray
            The IJK coordinates of peaks to convolve with the kernel.

        Returns
        --------
        sphere_coords : 2D numpy.ndarray
            All coordinates that fall within any sphere.
            Coordinates from overlapping spheres will appear twice.
        """
```

Hopefully this is correct.